frontend/src/lib/k8s/cluster.ts

```diff
@@ -679,37 +707,49 @@ export function makeKubeObject<T extends KubeObjectInterface | KubeEvent>(
         resourceAttrs['resource'] = this.pluralName;
       }
 
-      const spec = {
-        resourceAttributes: resourceAttrs,
-      };
+      // @todo: We should get the API info from the API endpoint.
+
+      // If we already have the group, version, and resource, then we can make the request
+      // without trying the API info, which may have several versions and thus be less optimal.
+      if (!!resourceAttrs.group && !!resourceAttrs.version && !!resourceAttrs.resource) {
+        return this.fetchAuthorization(resourceAttrs);
+      }
+
+      // If we don't have the group, version, and resource, then we have to try all of the
+      // API info versions until we find one that works.
+      const apiInfo = this.apiEndpoint.apiInfo;
+      for (let i = 0; i < apiInfo.length; i++) {
+        const { group, version, resource } = apiInfo[i];
+
+        if (!!resourceAttrs.resource) {
+          resourceAttrs.resource = resource;
+        }
+        if (!!resourceAttrs.group) {
+          resourceAttrs.group = group;
+        }
+        if (!!resourceAttrs.version) {
+          resourceAttrs.version = version;
+        }
+
+        let authResult;
 
-      const versions = ['v1', 'v1beta1'];
-      for (let i = 0; i < versions.length; i++) {
-        const version = versions[i];
         try {
-          return await post(
-            `/apis/authorization.k8s.io/${version}/selfsubjectaccessreviews`,
-            {
-              kind: 'SelfSubjectAccessReview',
-              apiVersion: `authorization.k8s.io/${version}`,
-              spec,
-            },
-            false
-          );
+          authResult = this.fetchAuthorization(resourceAttrs);
```

Comment:
Do you need to await here? Seems it's needed, otherwise the loop will just keep going.